app/models/name.rb

```diff
@@ -408,33 +408,95 @@ class Name < AbstractModel
     end
   end
 
-  scope :with_rank,
-        ->(rank) { where(rank: Name.ranks[rank]) if rank }
+  include CreatedUpdatedScopes
 
+  scope :of_lichens, -> { where(Name[:lifeform].matches("%lichen%")) }
+  scope :not_lichens, -> { where(Name[:lifeform].does_not_match("% lichen %")) }
+  scope :deprecated, -> { where(deprecated: true) }
   scope :not_deprecated, -> { where(deprecated: false) }
-
+  scope :with_description, -> { where.not(description_id: nil) }
+  scope :without_description, -> { where(description_id: nil) }
   ### Module Name::Spelling
   scope :with_correct_spelling, -> { where(correct_spelling_id: nil) }
-
-  # For a glitch discovered in the wild:
+  scope :with_incorrect_spelling, -> { where.not(correct_spelling_id: nil) }
   scope :with_self_referential_misspelling, lambda {
     where(Name[:correct_spelling_id].eq(Name[:id]))
   }
-
+  scope :with_synonyms, -> { where.not(synonym_id: nil) }
+  scope :without_synonyms, -> { where(synonym_id: nil) }
+  ### Module Name::Create
+  # Get list of Names that are potential matches when creating a new name.
+  # Takes results of Name.parse_name.  Used by NameController#create_name.
+  # Three cases:
+  #
+  #   1. group with author       - only accept exact matches
+  #   2. nongroup with author    - match names with correct author or no author
+  #   3. any name without author - ignore authors completely when matching names
+  #
+  # If the user provides an author, but the only match has no author, then we
+  # just need to add an author to the existing Name.  If the user didn't give
+  # an author, but there are matches with an author, then it already exists
+  # and we should just ignore the request.
+  #
+  scope :matching_desired_new_name, lambda { |parsed_name|
+    if parsed_name.rank == "Group"
+      where(search_name: parsed_name.search_name)
+    elsif parsed_name.author.empty?
+      where(text_name: parsed_name.text_name)
+    else
+      where(text_name: parsed_name.text_name).
+        where(author: [parsed_name.author, ""])
+    end
+  }
+  scope :matching_search_name,
+        ->(search_name) { where(search_name: search_name) }
   ### Module Name::Taxonomy
-  scope :with_classification_like,
+  scope :with_rank,
+        ->(rank) { where(rank: Name.ranks[rank]) if rank }
+  scope :with_rank_below,
+        ->(rank) { where(Name[:rank] < Name.ranks[rank]) }
+  scope :with_rank_classification_like,
         # Use multi-line lambda literal because fixtures blow up with "lambda":
         # NoMethodError: undefined method `ranks'
         #   test/fixtures/names.yml:28:in `get_binding'
         ->(rank, text_name) { # rubocop:disable Style/Lambda
           where(Name[:classification].matches("%#{rank}: _#{text_name}_%"))
         }
-
-  scope :with_name_like,
+  scope :with_name_like, # Note small diff w :text_name_like scope
```

Comment:
Yikes, this seems really misleading to me.  This is returning subtaxa of the given name!  But only for genera or below.  How about "subtaxa_for_genus_or_below"?  Or less precise but much shorter and may convey the right meaning anyway: "subtaxa_below_genus"?

Reply:
Are you referring to this scope returning subtaxa of the given name? I could not understand what it was doing differently!
Please confirm...
```ruby
  scope :with_name_like, # Note small diff w :text_name_like scope
        ->(text_name) { where(Name[:text_name].matches("#{text_name} %")) }
```

Reply:
This is a great case for named scopes being better than even simple explicit 'where' conditions!  The extra space will only occur in subtaxa.  Think of "Agaricus".  It's searching for names starting with "Agaricus" and having additional words (species, subspecies, etc.).  Agaricus itself will not match.  But Agaricus campestris, and so on, will.  But of course, it doesn't work on names above genus.  It won't match anything when run on Agaricaceae, for example.  For that you need to rely on the classification string being populated and including "Family: _Agaricaceae_".

Reply:
Ahaa! I get it now. Thank you